lib/basedriver/driver.js

```diff
@@ -51,6 +56,32 @@ class BaseDriver extends MobileJsonWireProtocol {
 
     // allow subclasses to have internal drivers
     this.managedDrivers = [];
+
+    // store event timings
+    this._eventHistory = {
+      commands: [] // commands get a special place
+    };
+  }
+
+  /*
+   * make eventHistory a property and return a cloned object so a consumer can't
+   * inadvertently change data outside of logEvent
+   */
+  get eventHistory () {
+    return _.cloneDeep(this._eventHistory);
+  }
+
+  /*
+   * API method for driver developers to log timings for important events
+   */
+  logEvent (eventName) {
+    if (eventName === "commands") {
+      throw new Error("Cannot log commands directly");
+    }
+    if (typeof eventName !== "string") {
+      throw new Error(`Invalid eventName ${eventName}`);
+    }
+    this._eventHistory[eventName] = Date.now();
```

Comment:
Do we want to only allow a single event of each type? What if we want to have events for WDA startup, which might have to happen multiple times in a session? Would that be the calling code's responsibility, to create reasonable non-clashing event names?